opencog/atomspace/atom_types.script

```diff
@@ -76,6 +76,9 @@ EXISTS_LINK <- BIND_LINK
 AVERAGE_LINK <- FORALL_LINK
 SATISFYING_SET_LINK <- BIND_LINK
 
+// Function constructor (predicate and schema)
+LAMBDA_LINK <- BIND_LINK
```

Comment:
Umm,

I think you got this backwards.  I think you want this:
```
LAMBDA_LINK <- ORDERED_LINK
BIND_LINK <- LAMBDA_LINK
FORALL_LINK <- LAMBDA_LINK
EXISTS_LINK <- LAMBDA_LINK
SATISFYING_SET_LINK <- LAMBDA_LINK
```
right? that makes bind-link a special case of the lambda, that the pattern matcher "understands" but has no other specific "meaning".

Also, perhaps BindLink should be renamed to GroundingLink or something like that.